src/pages/index.tsx

```diff
@@ -194,6 +239,10 @@ const Home: NextPage<any> = () => {
     }
   };
 
+  useEffect(() => {
```

Comment:
isn't this useEffect actually unnecessary? Rather than using a useEffect and rather than using this function to conditionally render, it should be cleaner and maybe more performance efficient if you just include both components using a ternary operator in the return method of index.tsx. eg: instead of {mapState && user && renderSidebar()} on line 257, should be {mapState && user && (sidebarState == 'explore' ? <ExploreSidebar ...etc/> : <RequestsSidebar ...etc/>)}.

The reason for this being is that the sidebars should already appropriately rerender when recommendations, favorites, or requests change, because they are already passed in as props, and so the useEffect is not necessary. 

Actually, what would be best is if both sidebars Sidebar were merged into a single component and then the conditional rendering occurs in that component. Although this technically works, I think putting both sidebars into a single component which takes in sidebarState as a prop is the cleanest solution and easiest to maintain. but maybe this could be a separate ticket.

Reply:
Yes, this just got left around from when i was experimenting with some weird behavior we were seeing. I'll remove both this and the commented out code.

Reply:
I'm also going to create a few new tickets on code consolidation on frontend so we can add your notes regarding Sidebar refactoring there.